### ISSUE TITLE
A different Culture could be being used when going from variable to interpolated string than you expect

### ISSUE THREAD

**Issue by stalhandske** [YarnSpinnerTool/YarnSpinner-Unity#140]
It seems like `{$someFloat}` evaluates depending on locale and can therefore give problems if used in a command.

Reproduce:
I'm on beta5.
On Windows 10.
My region is set to Denmark / Danish
In the visual novel example add the two lines
```
<<set $float = 1.1>>
<<Shake Eve, {$float}>>
```

This gives two errors
```
Can't run command Shake: ShakeSprite requires between 1 and 2 parameters, but 3 were provided.
UnityEngine.Debug:LogError (object)
Yarn.Unity.DialogueRunner:DispatchCommandToRegisteredHandlers (string) (at Library/PackageCache/dev.yarnspinner.unity@493931d540/Runtime/DialogueRunner.cs:894)
Yarn.Unity.DialogueRunner:DispatchCommandToRegisteredHandlers (Yarn.Command) (at Library/PackageCache/dev.yarnspinner.unity@493931d540/Runtime/DialogueRunner.cs:851)
Yarn.Unity.DialogueRunner:HandleCommand (Yarn.Command) (at Library/PackageCache/dev.yarnspinner.unity@493931d540/Runtime/DialogueRunner.cs:751)
Yarn.VirtualMachine:RunInstruction (Yarn.Instruction)
Yarn.VirtualMachine:Continue ()
Yarn.Dialogue:Continue ()
Yarn.Unity.DialogueRunner:ContinueDialogue () (at Library/PackageCache/dev.yarnspinner.unity@493931d540/Runtime/DialogueRunner.cs:1291)
Yarn.Unity.DialogueRunner:DialogueViewCompletedDismissal (Yarn.Unity.DialogueViewBase) (at Library/PackageCache/dev.yarnspinner.unity@493931d540/Runtime/DialogueRunner.cs:1395)
Yarn.Unity.DialogueRunner/<>c__DisplayClass86_0:<DismissLineFromViews>b__0 () (at Library/PackageCache/dev.yarnspinner.unity@493931d540/Runtime/DialogueRunner.cs:1380)
Yarn.Unity.DialogueViewBase:DismissLine (System.Action) (at Library/PackageCache/dev.yarnspinner.unity@493931d540/Runtime/Views/DialogueViewBase.cs:92)
Yarn.Unity.DialogueRunner:DismissLineFromViews (System.Collections.Generic.IEnumerable`1<Yarn.Unity.DialogueViewBase>) (at Library/PackageCache/dev.yarnspinner.unity@493931d540/Runtime/DialogueRunner.cs:1380)
Yarn.Unity.DialogueRunner:OnViewUserIntentNextLine () (at Library/PackageCache/dev.yarnspinner.unity@493931d540/Runtime/DialogueRunner.cs:1348)
Yarn.Unity.DialogueViewBase:ReadyForNextLine () (at Library/PackageCache/dev.yarnspinner.unity@493931d540/Runtime/Views/DialogueViewBase.cs:167)
Yarn.Unity.LineView:OnContinueClicked () (at Library/PackageCache/dev.yarnspinner.unity@493931d540/Runtime/Views/LineView.cs:399)
Yarn.Unity.LineView:Update () (at Library/PackageCache/dev.yarnspinner.unity@493931d540/Runtime/Views/LineView.cs:242)
```

and 

```
No Command <<Shake Eve, 1,1>> was found. Did you remember to use the YarnCommand attribute or AddCommandHandler() function in C#?
UnityEngine.Debug:LogError (object)
Yarn.Unity.DialogueRunner:HandleCommand (Yarn.Command) (at Library/PackageCache/dev.yarnspinner.unity@493931d540/Runtime/DialogueRunner.cs:776)
Yarn.VirtualMachine:RunInstruction (Yarn.Instruction)
Yarn.VirtualMachine:Continue ()
Yarn.Dialogue:Continue ()
Yarn.Unity.DialogueRunner:ContinueDialogue () (at Library/PackageCache/dev.yarnspinner.unity@493931d540/Runtime/DialogueRunner.cs:1291)
Yarn.Unity.DialogueRunner:DialogueViewCompletedDismissal (Yarn.Unity.DialogueViewBase) (at Library/PackageCache/dev.yarnspinner.unity@493931d540/Runtime/DialogueRunner.cs:1395)
Yarn.Unity.DialogueRunner/<>c__DisplayClass86_0:<DismissLineFromViews>b__0 () (at Library/PackageCache/dev.yarnspinner.unity@493931d540/Runtime/DialogueRunner.cs:1380)
Yarn.Unity.DialogueViewBase:DismissLine (System.Action) (at Library/PackageCache/dev.yarnspinner.unity@493931d540/Runtime/Views/DialogueViewBase.cs:92)
Yarn.Unity.DialogueRunner:DismissLineFromViews (System.Collections.Generic.IEnumerable`1<Yarn.Unity.DialogueViewBase>) (at Library/PackageCache/dev.yarnspinner.unity@493931d540/Runtime/DialogueRunner.cs:1380)
Yarn.Unity.DialogueRunner:OnViewUserIntentNextLine () (at Library/PackageCache/dev.yarnspinner.unity@493931d540/Runtime/DialogueRunner.cs:1348)
Yarn.Unity.DialogueViewBase:ReadyForNextLine () (at Library/PackageCache/dev.yarnspinner.unity@493931d540/Runtime/Views/DialogueViewBase.cs:167)
Yarn.Unity.LineView:OnContinueClicked () (at Library/PackageCache/dev.yarnspinner.unity@493931d540/Runtime/Views/LineView.cs:399)
Yarn.Unity.LineView:Update () (at Library/PackageCache/dev.yarnspinner.unity@493931d540/Runtime/Views/LineView.cs:242)
```

**Comment by stalhandske** [comment]
![image](https://user-images.githubusercontent.com/7137245/146207849-8f4b1cb4-185d-4ed1-91e2-d65bb59bd127.png)
This is the important one I think.

**Comment by desplesda** [comment]
This turned out to be a problem in the core Yarn Spinner compiler. I've fixed it in YarnSpinnerTool/YarnSpinner@36ee7f4f3067342cf4a63151d6757633a329e572 (v2.0.1) and have updated YarnSpinner for Unity to use this new version. Please see the [release notes for Yarn Spinner for Unity v2.0.1](https://github.com/YarnSpinnerTool/YarnSpinner-Unity/releases/tag/v2.0.1)!
